ablySpec/RealtimeClient.connection.swift

```diff
@@ -713,6 +713,49 @@ class RealtimeClientConnection: QuickSpec {
                 }
 
             }
+
+            // RTN12
+            context("close") {
+
+                // RTN12c
+                it("close move to CLOSED state if transport is abruptly closed") {
+                    let options = AblyTests.commonAppSetup()
+                    options.autoConnect = false
+                    let client = ARTRealtime(options: options)
+                    client.setTransportClass(TestProxyTransport.self)
+                    client.connect()
+
+                    let transport = client.transport as! TestProxyTransport
+                    var states: [ARTRealtimeConnectionState] = []
+
+                    waitUntil(timeout: testTimeout) { done in
+                        client.eventEmitter.on { state, errorInfo in
+                            switch state {
+                            case .Connected:
+                                states += [state]
+                                transport.closeAbruptly()
```

Comment:
Why are you waiting until it is connected for an abrupt close?  The spec says after calling `close`, if the transport is abruptly terminated then the connection simply moves from the `closing` state it would be in straight to the `closed` state